lib/view_component/collection.rb

```diff
@@ -10,19 +10,23 @@ def render_in(view_context, &block)
       @component.compile(raise_errors: true)
       @component.validate_collection_parameter!(validate_default: true)
 
-      @collection.map do |item|
+      @collection.each_with_index.map do |item, index|
```

Comment:
I wonder if we can continue to use `.map` here, and rely on `iterator.first`? https://api.rubyonrails.org/classes/ActionView/PartialIteration.html

Reply:
That makes sense to me, I didn't know it existed :D

Reply:
You can also do `@collection.map.with_index do ...`